ts/src/gemini.ts

```diff
@@ -255,22 +270,139 @@ export default class gemini extends Exchange {
                     'filecoin': 'FIL',
                     'dogecoin': 'DOGE',
                     'tezos': 'XTZ',
+                    'avalanche': 'AVALANCHE_X',
+                    'solana': 'SOLANA',
+                    'cosmos': 'COSMOS',
+                    'polkadot': 'POLKADOT',
                 },
                 'networks': {
                     'BTC': 'bitcoin',
+                    'ETH': 'ethereum',
                     'ERC20': 'ethereum',
                     'BCH': 'bitcoincash',
                     'LTC': 'litecoin',
+                    'ZCASH': 'zcash',
                     'ZEC': 'zcash',
+                    'FILECOIN': 'filecoin',
                     'FIL': 'filecoin',
+                    'DOGECOIN': 'dogecoin',
                     'DOGE': 'dogecoin',
+                    'TEZOS': 'tezos',
                     'XTZ': 'tezos',
+                    'AVALANCHE_X': 'avalanche',
+                    'SOLANA': 'solana',
+                    'COSMOS': 'cosmos',
+                    'POLKADOT': 'polkadot',
                 },
                 'nonce': 'milliseconds', // if getting a Network 400 error change to seconds
             },
         });
     }
 
+    async fetchCurrencies (params = {}) {
+        /**
+         * @method
+         * @name gemini#fetchCurrencies
+         * @description fetches all available currencies on an exchange
+         * @param {object} params extra parameters specific to the endpoint
+         * @returns {object} an associative dictionary of currencies
+         */
+        return await this.fetchCurrenciesFromWeb (params);
+    }
+
+    async fetchCurrenciesFromWeb (params = {}) {
+        /**
+         * @method
+         * @name gemini#fetchCurrenciesFromWeb
+         * @description fetches all available currencies on an exchange
+         * @param {object} params extra parameters specific to the endpoint
+         * @returns {object} an associative dictionary of currencies
+         */
+        const data = await this.fetchWebEndpoint ('fetchCurrencies', 'webExchangeGet', true, '="currencyData">', '</script>');
+        if (data === undefined) {
+            return undefined;
+        }
+        //
+        //    {
+        //        "tradingPairs": [
+        //            [ "BTCAUD", 2, 8, "0.00001", 10, true ],
+        //            ...
+        //        ],
+        //        "currencies": [
+        //            [ "ORCA", "Orca", 204, 6, 0, 6, 8, false, null, "solana" ],
```

Comment:
@ttodua question, did you find the meaning of each value when fetching it through the web endpoint?

Reply:
this was a relatively old PR (several months ago), which I have reviewed a few days ago. but what I remember, yes, I had checked & compared to UI functionalities and found out, otherwise I hadn't done that by random guess

Reply:
but wait, i will recheck

Reply:
as wrote in discord, i matched them